pkg/controllers/nodetemplate/suite_test.go

```diff
@@ -362,4 +364,166 @@ var _ = Describe("AWSNodeTemplateController", func() {
 			Expect(nodeTemplate.Status.SecurityGroups).To(BeNil())
 		})
 	})
+	Context("AMI Status", func() {
+		It("Should expect no errors when AMI selector is not in the AWSNodeTemplate", func() {
+			nodeTemplate.Spec.AMISelector = nil
+			ExpectApplied(ctx, env.Client, nodeTemplate)
+			ExpectReconcileSucceeded(ctx, controller, client.ObjectKeyFromObject(nodeTemplate))
+			nodeTemplate = ExpectExists(ctx, env.Client, nodeTemplate)
+			amiFamily := amifamily.GetAMIFamily(nodeTemplate.Spec.AMIFamily, &amifamily.Options{})
+			amis, _ := awsEnv.AMIProvider.GetAMIWithRequirements(ctx, nodeTemplate, amiFamily)
```

Comment:
These tests are somewhat short-circuiting a true expectation. Imagine, for example, that there is a bug in your `GetAMIWithRequirements` function. Though your test will pass when this is the case, the actual logic may be faulty. Generally, I think it's more accurate to have strong, static expectations. Like saying that this is the exact set of requirements and AMI ids that should be returned. That ensures that your logic is sound throughout rather than checking against other generated output.